appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java

```diff
@@ -98,29 +97,46 @@
   static final int ADDITIONAL_HTTP_TIMEOUT_BUFFER_MS = 1000;
 
   protected int defaultTimeoutMs;
+
   protected final ExecutorService executor;
 
   protected final HttpClient httpclient;
 
-  final IdleConnectionMonitorThread monitorThread;
+  //final IdleConnectionMonitorThread monitorThread;
 
-  private static ClientConnectionManager createConnectionManager() {
-    PoolingClientConnectionManager connectionManager = new PoolingClientConnectionManager();
-    connectionManager.setMaxTotal(VmApiProxyEnvironment.MAX_CONCURRENT_API_CALLS);
-    connectionManager.setDefaultMaxPerRoute(VmApiProxyEnvironment.MAX_CONCURRENT_API_CALLS);
-    return connectionManager;
+  public VmApiProxyDelegate() {
+    this(createHttpClient());
   }
 
-  public VmApiProxyDelegate() {
-    this(new DefaultHttpClient(createConnectionManager()));
+  private static HttpClient createHttpClient() {
+    try {
+      HttpClientTransportOverHTTP transport = new HttpClientTransportOverHTTP();
+      // FIXME what sort of customize SslContextFactory we want to use here??
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      HttpClient httpClient = new HttpClient( transport, sslContextFactory );
+      httpClient.setMaxConnectionsPerDestination( VmApiProxyEnvironment.MAX_CONCURRENT_API_CALLS );
+      httpClient.setConnectTimeout( ADDITIONAL_HTTP_TIMEOUT_BUFFER_MS );
```

Comment:
@meltsufin AFAICS the connection timeout was alway set to DEFAULT_RPC_TIMEOUT_MS. So I will add to have similar behaviour.